### PR TITLE
update libuv

### PR DESCRIPTION
**Pull request by vtjnash**
https://github.com/JuliaLang/julia/pull/32211

### CODE REVIEW

**Comment by vtjnash** [comment]
bump

File: LibUV/build_tarballs.jl [hunk 1]
```diff
@@ -1,12 +1,12 @@
 using BinaryBuilder
 
 name = "LibUV"
-version = v"1.24.0"
+version = v"2-1.24.0-1"
```
Comment (staticfloat) [inline LibUV/build_tarballs.jl]: I'm confused by this version number?  What does this mean?

Comment (vtjnash) [reply LibUV/build_tarballs.jl]: The commit hash below corresponds to the libuv version "2.0.0+1.24.0-julia-1", where `2.0.0` is the major/minor/patch (prerelease)version, `+1.24.0` is the bugfix id for the branch version, `-julia` is the patch set id, and `-1` is the distinguisher to indicate the first build in the series. I shortened it a bit here to just include the numbers that might be currently relevant for Julia.

Comment (staticfloat) [reply LibUV/build_tarballs.jl]: We don't want to encode build numbers into the `version`; this version should only identify the source; the build version numbers get added on later.  I'm also not certain we want the `2-` at the beginning; shouldn't we just leave this at `1.24.0`?

Comment (vtjnash) [reply LibUV/build_tarballs.jl]: The version number is 2, `1.24.0-julia` is just the name string of the current head of the branch

Comment (staticfloat) [reply LibUV/build_tarballs.jl]: Does that mean that when the source next changes, we'll have a version string of `3` with a name string of `1.24.0-julia`?

Comment (vtjnash) [reply LibUV/build_tarballs.jl]: No, it’ll probably be at version 2.0 for years. The next version would usually be 1.24.0-julia-2 then -3, etc.

In this particular case, as soon as this PR is done, I’ll be replacing it with 2+1.29.1-julia-1, but that’s unusual to have the updates so close together.

**Comment by staticfloat** [comment]
I assume you want new binaries of this built, yes?

**Comment by vtjnash** [comment]
yeah, I thought that was automatic

**Comment by staticfloat** [comment]
Unfortunately not yet; but we're getting closer.  In the meantime, when nothing goes wrong, it's just two command-line invocations for me, so not bad.  Here you go: https://github.com/JuliaPackaging/Yggdrasil/releases/tag/LibUV-v2%2B1.24.0-julia%2B0

**Pull request merged**